doc/declarative-state-allocation/rds.md

```diff
@@ -0,0 +1,146 @@
+        allocateDatabase:
+          storage: 400
+          instanceClass: db.t3.medium
+        name: chainsync-qa-mainnet
+    type: webservice
+```
+Verifying resource:
+
+```shell
+# Please note that CrossPlane objects are not namespace scoped. This is not a bug!
+% kubectl get rdsinstance
+NAME                   READY   SYNCED   STATE       ENGINE     VERSION   AGE
+chainsync-qa-mainnet   True    True     available   postgres   15.4      11h
+# Ready and Synced = True, State is available
+# Everything is Okay!
+```
+
+Usually it takes up to 30 minutes to complete. Some relevant notes about the new instances:
+
+1. We are using a "Security by default" approach:<br>
+    1.1. Databases are not publicly available.<br>
+    1.2. In-transit encryption is supported.<br>
+    1.3. At-rest encryption is enabled.<br>
+    1.4. Passwords are randomly generated by AWS and saved in Kubernetes secrets.<br>
+    1.5. Automated backups are enabled.<br>
+2. Automatic minor version upgrades is disabled by default.
+3. Performance Insights is enabled by default.
+4. Multi-AZ is enabled by default.
+5. Default region is us-east-1 (North Virginia).
+
+This trait also has support for setting some optional properties:
+
+| Property  |  Description  |
+|---|---|
+| `backupRetentionPeriod` | Defines a backup retention period (Number of days)<br>|
+| `preferredBackupWindow` | Defines backup window schedule |
+| `preferredMaintenanceWindow` | Defines maintenance window schedule |
+| `iops` | Overrides IOPS settings |
+| `deletionProtection` | Enable deletion protection feature |
+| `customParameters` | Update RDS [parameter group](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.PostgreSQL.CommonDBATasks.Parameters.html)<br>*A list of all available parameters can be found [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.PostgreSQL.CommonDBATasks.Parameters.html#Appendix.PostgreSQL.CommonDBATasks.Parameters.parameters-list)* |
+
+**⚠️ IMPORTANT**!
+
+Without `deletionProtection`, the RDS instance is tied to the parent Application lifecycle so if the Application is deleted (or just recycled through VelaUX, for example), the RDS instance is deleted as well.
+
+A note about backups
+---------------
+
+As said above, the automated backups are enabled by default. If `backupRetentionPeriod` is not set up, we will assume a default value of 1, which means that all instances will keep exactly 1 snapshot updated. If `backupRetentionPeriod` is set to zero (0), no automated backup will be taken.
+
+
+Using the instance
+---------------
+
+After applying the YAML from the example above, we will have a new RDS instance after some minutes and then, it will be time to use it. The ***rds-instance*** trait also has a Patch feature that will update the deployment to ensure that it has all the necessary environment variables to use the RDS instance.
+
+If we have another Application that needs to use the same RDS instance, we can use as such:
```

Comment:
We should track some followup work for access control... Ideally one team should not be able to freely request access to anothers' db.

Reply:
@shlevy Do we have more than one team using a single project/namespace?
I am asking it because this trait will only allow access to databases allocated in the same project/namespace.

Reply:
Project/namespace scoping is a good first-pass approach, sounds good.